extra_data/tests/test_reader_mockdata.py

```diff
@@ -990,3 +991,273 @@ def test_run_metadata_no_trains(mock_scs_run):
     sel = run.select_trains(np.s_[:0])
     md = sel.run_metadata()
     assert md['dataFormatVersion'] == '1.0'
+
+
+def test_with_aliases(mock_sa3_control_data, mock_sa3_control_aliases):
+    run = H5File(mock_sa3_control_data).with_aliases(mock_sa3_control_aliases)
+
+    def assert_equal_keydata(kd1, kd2):
+        assert isinstance(kd1, KeyData)
+        assert isinstance(kd2, KeyData)
+        assert kd1.source == kd2.source
+        assert kd1.key == kd2.key
+        assert kd1.train_ids == kd2.train_ids
+
+    # Test whether source alias yields identical SourceData.
+    assert run.alias['sa3-xgm'] is run['SA3_XTD10_XGM/XGM/DOOCS']
+
+    # Test whether source alias plus literal key yields equal KeyData.
+    assert_equal_keydata(
+        run.alias['sa3-xgm', 'pulseEnergy.wavelengthUsed'],
+        run['SA3_XTD10_XGM/XGM/DOOCS', 'pulseEnergy.wavelengthUsed'])
+
+    # Test whether key alias yields equal KeyData.
+    assert_equal_keydata(
+        run.alias['hv'],
+        run['SA3_XTD10_XGM/XGM/DOOCS', 'pulseEnergy.wavelengthUsed'])
+
+    # Test undefined aliases.
+    with pytest.raises(AliasError):
+        run.alias['foo']
+        run.alias['foo', 'bar']
+
+    # Test using a literal key with a key alias.
+    with pytest.raises(ValueError):
+        run.alias['hv', 'pressure']
+
+    # Test using an existing source alias for a non-existing source.
+    with pytest.raises(SourceNameError):
+        run.alias['bogus-source']
+
+    # Test using an existing key alias for a non-existing key.
+    with pytest.raises(PropertyNameError):
+        run.alias['bogus-key']
```

Comment:
I was initially somewhat surprised that you could create invalid aliases and get an error only when accessing them.

I think that probably makes sense - it allows someone to define one big collection of aliases where runs will only use a subset of the relevant sources. But it would be nice to know about typos in the alias definitions before someone tries to use them. Maybe we could include a method like `run.alias.check()` to tell you which aliases point to missing sources or keys?

This doesn't have to be part of this PR, though - maybe we want to think more about how best to expose that information.

Reply:
This would be possible right now by `run.only_aliases(strict=True)`, albeit you wouldn't get a summary, but the included `.select()` call would fail at the first missing source/key.